src/test/deploy/hosting/convertConfig.spec.ts

```diff
@@ -300,3 +360,131 @@ describe("convertConfig", () => {
+      input: { rewrites: [{ glob: "/foo", function: "foofn" }] },
+      errorString: "More than one backend found for function name",
+      payload: {
+        functions: {
+          default: {
+            wantBackend: backend.of(
+              {
+                id: "foofn",
+                project: "my-project",
+                entryPoint: "foofn",
+                runtime: "nodejs14",
+                region: "us-central1",
+                platform: "gcfv2",
+                httpsTrigger: {},
+              },
+              {
+                id: "foofn",
+                project: "my-project",
+                entryPoint: "foofn",
+                runtime: "nodejs14",
+                region: "europe-west2",
+                platform: "gcfv2",
+                httpsTrigger: {},
+              }
+            ),
+            haveBackend: backend.empty(),
+          },
+        },
+      },
+      finalize: true,
+    },
+    {
+      name: "multiple v1 endpoints are found for function and functionRegion is not specified",
+      input: { rewrites: [{ glob: "/foo", function: "foofn" }] },
+      errorString: "More than one backend found for function name",
+      payload: {
+        functions: {
+          default: {
+            wantBackend: backend.of(
+              {
+                id: "foofn",
+                project: "my-project",
+                entryPoint: "foofn",
+                runtime: "nodejs14",
+                region: "us-central1",
+                platform: "gcfv1",
+                httpsTrigger: {},
+              },
+              {
+                id: "foofn",
+                project: "my-project",
+                entryPoint: "foofn",
+                runtime: "nodejs14",
+                region: "europe-west2",
+                platform: "gcfv1",
+                httpsTrigger: {},
+              }
+            ),
+            haveBackend: backend.empty(),
+          },
+        },
+      },
+      finalize: true,
+    },
+    {
+      name: "multiple existing v1 function regions are found for rewrite and no region is specified",
+      input: { rewrites: [{ regex: "/foo$", function: "foofn" }] },
+      context: {
+        loadedExistingBackend: true,
+        existingBackend: {
+          endpoints: {
+            "us-central1": {
+              foofn: {
+                id: "foofn",
+                region: "us-central1",
+                platform: "gcfv1",
+                httpsTrigger: true,
+              },
+            },
+            "asia-northeast2": {
+              foofn: {
+                id: "foofn",
+                region: "asia-northeast2",
+                platform: "gcfv1",
+                httpsTrigger: true,
+              },
+            },
+          },
+        },
+      },
+      errorString: "More than one backend found for function name",
+      finalize: true,
+    },
+  ];
+
+  for (const {
+    name,
+    context = DEFAULT_CONTEXT,
+    input,
+    payload = DEFAULT_PAYLOAD,
```

Comment:
IMO, having the DEFAULT_* variables set here is an anti-pattern. You want all the context for each test case _in the test case_, so they should be defined above in the cases. Please go ahead and make this change.

(this is also why I was very confused why payload was being added in the other set of tests - I didn't understand why it's _now_ returning a payload, instead of it just being changed)

Reply:
Fixed.